src/pricecypher/contracts/ScopeScript.py

```diff
@@ -0,0 +1,35 @@
+from abc import ABC, abstractmethod
+from typing import Any, Optional
+
+from pricecypher.contracts.Script import Script
+
+
+class ScopeScript(Script, ABC):
+    """
+    The abstract ScopeScript class serves as an interaction contract such that by extending it with its
+        methods implemented, a script can be created that calculates values for some scope for transactions,
+        which can then be used in a generalized yet controlled setting.
+    """
+
+    def execute(self, business_cell_id: Optional[int], bearer_token: str, user_input: dict[Any: Any]) -> Any:
+        # Executing a scope-script like a normal script:
+        # Attempt to extract the transaction IDs and continue as normal
+        transaction_ids: list[int] = user_input['transaction_ids']
+        return self.execute_scope_script(business_cell_id, bearer_token, transaction_ids)
+
+    @abstractmethod
+    def execute_scope_script(
+        self,
+        business_cell_id: Optional[int],
+        bearer_token: str,
+        transaction_ids: list[int]
```

Comment:
Should the rest of the user_input not be included here as well?

Reply:
I don't think we'd really want it to take in anything else, at least not with the current plans. If something is a scope-calculating script, it should be able to do its calculations with just these fields and the settings and config values (and potentially externally fetched data of course), without needing additional information sent along with the script call.